### PR TITLE
Merchant bulk buy function

### PR DESCRIPTION
**Pull request by Nerrathiel**
Adds additional function to the merchant software - Bulk Buy.

Activating this function prompts the user to enter a number. 
   - If text is SOMEHOW entered, returns IC error message.
   - If number lesser than 1 is entered, returns IC error message.
   - Expects numbers equal or greater to 1.

Function then executes standard money-purchase function using a loop. This loop executes X amount of times where X is the number user entered, ensuring they purchase the amount of items they want to buy up to as much money as they actually have
   - This makes sure they cannot abuse this to buy more things than they can afford.

This function greatly reduces lag caused by server spawning excess amount of items and makes merchant much more pleasant to play.

Compiled without errors, tested on up-to-date private server, worked as expected.

### CODE REVIEW

**Comment by Nerrathiel** [comment]
!review

File: code/modules/merchant/merchant_programs.dm [hunk 1]
```diff
@@ -81,6 +81,28 @@
...
+	var/BulkCounter = 0
+	var/BulkAmount = input("How many?") as num
+	if(istext(BulkAmount))
+		last_comms = "ERROR: NUMBER EXPECTED"
+		return 
+	if(BulkAmount < 1)
+		last_comms = "ERROR: NUMBER 1 OR GREATER EXPECTED"
+		return
+	if(pad)
+		while(BulkCounter < BulkAmount)
```
Comment (NonQueueingMatt) [inline code/modules/merchant/merchant_programs.dm]: This needs to be a for loop.

File: code/modules/merchant/merchant_programs.dm [hunk 2]
```diff
@@ -81,6 +81,26 @@
...
+/datum/computer_file/program/merchant/proc/bulk_offer(var/datum/trader/T, var/num)
+	var/BulkAmount = input("How many items? (1-50)") as num
+	if(istext(BulkAmount))
+		last_comms = "ERROR: NUMBER EXPECTED"
+		return 
+	if(BulkAmount < 1 || BulkAmount > 50)
+		last_comms = "ERROR: NUMBER BETWEEN 1 AND 50 EXPECTED"
+		return
+	if(pad)
+		for(var/BulkCounter = 0, BulkCounter<BulkAmount, BulkCounter++)
```
Comment (NonQueueingMatt) [inline code/modules/merchant/merchant_programs.dm]:
```suggestion
		for(var/BulkCounter = 0, BulkCounter < BulkAmount, BulkCounter++)
```

File: code/modules/merchant/merchant_programs.dm [hunk 3]
```diff
@@ -81,6 +81,26 @@
 			bank -= response
 		return
 	last_comms = "PAD NOT CONNECTED"
+	
+/datum/computer_file/program/merchant/proc/bulk_offer(var/datum/trader/T, var/num)
+	var/BulkAmount = input("How many items? (1-50)") as num
```
Comment (Sindorman) [inline code/modules/merchant/merchant_programs.dm]: You could make it accept 0 as cancelling input. In that case the loop will just not execute and you just need to change line 90 if check to be `if(BulkAmount < 0 || BulkAmount > 50)`

Comment (Nerrathiel) [reply code/modules/merchant/merchant_programs.dm]: Implemented. Resolving

**Pull request merged**